### DISCUSSION TITLE
How to set the minimum size

### DISCUSSION THREAD

**Discussion by bizehao** [microsoft/WindowsAppSDK#2797] [Q&A] [answered]
How to set the minimum size in winui3,Limit the minimum stretch size

**Comment by mikebattista** [comment, marked as answer]
@dotmorten's [WinUIEx](https://github.com/dotMorten/WinUIEx) makes it really easy to set MinWidth and MinHeight on the Window in XAML.

The latest [Template Studio 5.2](https://github.com/microsoft/WindowsAppSDK/discussions/2795) update does just this.

**Reply by dotMorten** [reply]
Doc for reference: https://dotmorten.github.io/WinUIEx/concepts/WindowManager.html

**Reply by Balkoth** [reply]
While i appreciate that @dotMorten provides an easy way to do this, i question why this is not available on the base window class. Winforms and WPF did this years ago without hustle and now one has to use third party code for it or dive into unmanaged code to get basic windowing functionality to work.

**Reply by jonwis** [reply]
@rkarman - seems like a reasonable Windowing ask...

**Reply by rkarman** [reply]
Yup, agree, Min/Max size properties/methods are items we've discussed in the past as well and have on the backlog to go add to the AppWindow. I've made sure that they are up for discussion for the 1.3 planning cycle.

**Reply by dotMorten** [reply]
Also please consider adding more scuff directly to the Window class. It's weird you have to go get the app window first (which isn't even a simple property). I get appwindow is for windows in general but most of those other window frameworks already have these properties. WinUI is what is missing simplified window settings, not AppWindow

**Reply by GF-Huang** [reply]
I don't know why not make these property into class `Window`, why we need `AppWindow`, can someone explains why?
